zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/slack/StartUpAlerterImpl.java

```diff
@@ -40,18 +50,46 @@ public StartUpAlerterImpl(SlackClient slack, List<String> channels) {
     StartUpAlerterImpl(SlackClient slack, List<String> channels, boolean queueLocked,
                        boolean lockedNotificationSent) {
         this.slack = slack;
-        this.channels = channels;
         this.queueLocked = queueLocked;
         this.lockedNotificationSent = lockedNotificationSent;
-        this.messages = new ArrayList<>();
+
+        this.queueLockedAlerts = channels.stream()
+                .map(c -> newQueueLockedAlertTask(c))
+                .collect(Collectors.toList());
+
+        this.queueUnlockedAlerts = new ArrayList<>();
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            info().log("shutting down slack thread executor service");
+            executorService.shutdown();
+        }));
     }
 
     @Override
     public void queueLocked() {
         synchronized (MUTEX) {
             if (queueLocked && !lockedNotificationSent) {
-                channels.stream().forEach(c -> messages.add(publishingQueueLocked(c)));
+
+                List<Future<PostMessage>> futures = null;
+                try {
+                    futures = executorService.invokeAll(queueLockedAlerts);
+                } catch (Exception ex) {
+                    error().exception(ex).log("error sending publishing queue locked slack notification");
+                    return;
+                }
+
                 this.lockedNotificationSent = true;
```

Comment:
Just wondering if this is needed anymore? The notification(s) are sent if `queueUnlockedAlerts` is no-empty. In any case it isn't true if all fail as this wont be identified until the futures are consumed in the next block.

Reply:
Good point. This was a hangover from the first iteration, i'll remove it. The same can be determined by queueUnlockedAlerts being non empty.